distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/InstallPluginActionTests.java

```diff
@@ -404,6 +408,60 @@ public void testSomethingWorks() throws Exception {
         assertPlugin("fake", pluginDir, env.v2());
     }
 
+    public void testSlowSignatureVerificationMessage() throws Exception {
+        String url = "https://artifacts.elastic.co/downloads/elasticsearch-plugins/analysis-icu/analysis-icu-"
+            + Build.CURRENT.qualifiedVersion()
+            + ".zip";
+        final MessageDigest digest = MessageDigest.getInstance("SHA-512");
+
+        InstallPluginAction action = makeActionPluginThatDownloads(
+            "analysis-icu",
+            url,
+            null,
+            false,
+            ".sha512",
+            checksumAndFilename(digest, url),
+            newSecretKey(),
+            this::signature
+        );
+
+        final InstallPluginAction spied = spy(action);
+
+        // make the slow verification acceptable delay artificially low for testing
+        doReturn(100L).when(spied).acceptableSignatureVerificationDelay();
+
+        doAnswer(i -> {
+            Thread.sleep(200); // sleep 100ms more than acceptable delay
```

Comment:
Can we just block here indefinitely? I'm concerned for flakiness if this completes within the same slice, and the wrong thread unblocks first, that this could appear "ok" and cause the test to fail.

Reply:
Good point!